### PR TITLE
fix out of tree doc build

### LINKED ISSUE #20457: Out-of-tree install fails due to docs
Running `make install` from an out-of-tree build seems to fail in the documentation building phase. It appears that the `Makefile` searches the docs `make.jl` script in the build not source directory.
[Gist with the output of `make install`](https://gist.github.com/mortenpi/03bfa592f9664b1cea7df9d3ff51c33c).

### PR DESCRIPTION
**Pull request by tkelman**
closes #20457

### CODE REVIEW

File: doc/make.jl [hunk 1]
```diff
@@ -113,7 +113,7 @@ const PAGES = [
 ]
 
 makedocs(
-    build     = "_build/html/en",
+    build     = joinpath(pwd(), "_build/html/en"),
```
Comment (tkelman) [inline doc/make.jl]: @mortenpi this seems like it should be the implicit default behavior, what's up with this being needed?

Comment (mortenpi) [inline doc/make.jl]: The `build` is relative to `root`, which defaults to the script's directory (`Base.source_dir()`), which will be in the source not build directory.

**Comment by mortenpi** [comment]
Thanks for tackling this @tkelman!

The building part now works, but it now fails in `deploydocs` -- it tries to figure out the commit with `git rev-parse` but it gets run in `pwd()`, which is not a repo ([log here](https://gist.github.com/mortenpi/03bfa592f9664b1cea7df9d3ff51c33c#file-make-install-log-2)). But this is actually something that we should fix in Documenter I think. Probably have a `cd(root) do ... end` around the `git rev-parse`, I'll try that.

**Comment by tkelman** [comment]
is deploydocs needed in a local build? ideally it should be possible to build html from a non git copy, providing version info to Documenter through some other means

**Comment by mortenpi** [comment]
The way I see it is that the assumption is that `deploydocs` gets called
even on non-deploy builds. It's `deploydocs`' job to figure out whether it
should do something or not. If not then it should just do nothing and quit
silently.

It's a good point though that my solution will still fail on non-git builds.

In any case, the `make.jl` script checks for `"deploy" in ARGS` and the
Makefile always adds it, even on user builds. Should that be dropped maybe
and explicitly added only on CI workers?

cc @MichaelHatherly, in case you have any thoughts.

**Comment by mortenpi** [comment]
To actually answer the question -- no, `deploydocs` is certainly not _necessary_ on local builds. But it should just exit without an error in case it does get called, and I PRd a fix to Documenter for that.

**Comment by tkelman** [comment]
@mortenpi should we merge this now or wait for some additional changes?

osx timeout probably unrelated, log backed up to https://gist.github.com/16cfca62b5dfc004f1be873330ac65d8 and restarted

**Comment by mortenpi** [comment]
One minor thing I noticed was that running `make install` again after Documenter fails will not try to build the docs again without me deleting the `doc/_build` directory first. Not sure if that could be fixed easily though.. clean up `_build` if Documenter gives a non-zero?

Besides that, the build works fine for me with this, I don't think there's anything else needed here. To get it completely working the `deploydocs` fix needs to be merged to Documenter and a new version tagged, but that shouldn't hold this up.

**Comment by tkelman** [comment]
I think Mike said he'd be traveling for a while, so will probably have a bit of a backlog to catch up on whenever he resurfaces.

**Pull request merged**